setup.sh

```diff
@@ -1,9 +1,21 @@
+# Use '-i' when sourcing this file to attempt to reset $PATH to system defaults (i.e., based on /etc/profile),
+# then load any modules (e.g., python) critical for loading and working with Spack.
+# This may have unwanted/unexpected behavior on some systems, so use carefully.
+if [ "$1" == "-i" ]; then
```

Comment:
Intersting approach. Let's try this out for the upcoming release. I always thought that we can trim down the instructions for the tier-1 platforms ("load these modules for building spack environments") substantially, which falls into the same category as what you are trying to do here.